client/components/domains/register-domain-step/index.jsx

```diff
@@ -31,19 +32,53 @@ var wpcom = require( 'lib/wp' ).undocumented(),
 	abtest = require( 'lib/abtest' ).abtest;
 
 // max amount of domain suggestions we should fetch/display
-var SUGGESTION_QUANTITY = 10,
+const SUGGESTION_QUANTITY = 10,
 	INITIAL_SUGGESTION_QUANTITY = 2;
 
+const analytics = analyticsMixin( 'registerDomain' );
+
+let searchQueue = [],
+	searchStackTimer = null;
+
+function processSearchQueue() {
+	var q = searchQueue.slice();
+	window.clearTimeout( searchStackTimer );
+	searchStackTimer = null;
+	searchQueue = [];
+
+	outerLoop:
+		for ( let i = 0; i < q.length; i++ ) {
+			for ( let k = i + 1; k < q.length; k++ ) {
+				if ( startsWith( q[ k ].query, q[ i ].query ) ) {
+					continue outerLoop;
+				}
+			}
+			reportSearch( q[ i ] );
+		}
+}
+
+function reportSearch( { query, section } ) {
+	analytics.recordEvent( 'searchFormSubmit', query, section );
+}
+
+function enqueueSearch( search ) {
+	searchQueue.push( search );
+	if ( searchStackTimer ) {
+		window.clearTimeout( searchStackTimer );
+	}
+	searchStackTimer = window.setTimeout( processSearchQueue, 10000 );
+}
+
 var RegisterDomainStep = React.createClass( {
-	mixins: [ analyticsMixin( 'registerDomain' ) ],
+	mixins: [ analytics ],
 
 	propTypes: {
 		cart: React.PropTypes.object,
 		onDomainsAvailabilityChange: React.PropTypes.func,
 		products: React.PropTypes.object.isRequired,
 		selectedSite: React.PropTypes.oneOfType( [ React.PropTypes.object, React.PropTypes.bool ] ),
 		basePath: React.PropTypes.string.isRequired,
-		suggestion: React.PropTypes.string,
+		suggestion: React.PropTypes.string
```

Comment:
This is probably the result of some automatic IDE formatting, but I'd prefer if you didn't remove trailing commas like these (especially if they are already there). They make adding new lines to arrays, objects, etc. much easier (the diff is clearer and contained to the line actually being modified). We have a rule for that for the backend code, but, surprisingly, not for the frontend :/